### ISSUE TITLE
Fullscreen apps is not launching in fullscreen (Wayland) 

### ISSUE THREAD

**Issue by SDD701** [qtile/qtile#4900]
### Issue description

When I launch a fullscreen app (some game that is set to fullscreen in the settings),  it's not launching in fullscreen on wayland.  Autofullscreen option is untouched (true according to docs).  I'm using Gentoo if it matters. 

### Version

0.26.0

### Backend

Wayland (experimental)

### Config

```python
from libqtile import qtile          # For various settings between X11 and Wayland 
from libqtile.config import Key     # For setting keybindings
from libqtile.lazy import lazy      # For perform actions on qtile itself (qtile commands API)
from libqtile import hook           # For hooks (events)
from os import system as shell      # For calling complicated shell commands
from subprocess import Popen as spr # For scrips that is stopping qtile from booting/working

# Preset
from libqtile import bar, layout, widget
from libqtile.config import Click, Drag, Group, Match, Screen


# Backend specific settings
if qtile.core.name == "x11":
    # Some config
    print('X11')
elif qtile.core.name == "wayland":
    # Some config
    print('Wayland')


# Declare constants
mod             = "mod4"    # [Win] Main key, that is used for all bindings
mod_alternative = "mod1"    # [Alt] If used with mod, do alternative action (even more bindings)

# Set keybindings
keys = [
    # Launch apps on a keybinding
    Key([mod], "q", lazy.spawn("kitty")),
    Key([mod], "r", lazy.spawn("wofi --show drun")),

    # Control windows
    Key([mod], "c", lazy.window.kill()),
    Key([mod], "o", lazy.window.toggle_fullscreen()),

    # Control WM
    Key([mod, mod_alternative], "r", lazy.reload_config()),
    Key([mod], "space", lazy.widget["keyboardlayout"].next_keyboard()),
]


# Setup groups
groups = [Group(i) for i in "123456789"]
for i in groups:
    keys.extend(
        [
            # mod + group number = switch to group
            Key(
                [mod],
                i.name,
                lazy.group[i.name].toscreen(),
                desc="Switch to group {}".format(i.name),
            ),
            # mod + shift + group number = switch to & move focused window to group
            Key(
                [mod, "shift"],
                i.name,
                lazy.window.togroup(i.name, switch_group=True),
                desc="Switch to & move focused window to group {}".format(i.name),
            ),
            # Or, use below if you prefer not to switch to that group.
            # # mod + shift + group number = move focused window to group
            # Key([mod, "shift"], i.name, lazy.window.togroup(i.name),
            #     desc="move focused window to group {}".format(i.name)),
        ]
    )


layouts = [
    layout.Columns(border_focus_stack=["#d75f5f", "#8f3d3d"], border_width=4),
    layout.Max(),
    layout.Stack(num_stacks=2),
    layout.Bsp(),
    layout.Matrix(),
    layout.MonadTall(),
    layout.MonadWide(),
    layout.RatioTile(),
    layout.Tile(),
    layout.TreeTab(),
    layout.VerticalTile(),
    layout.Zoomy(),
]

screens = [
    Screen(
        top=bar.Bar(
            [
                widget.CurrentLayout(),
                widget.GroupBox(),
                widget.Prompt(),
                widget.WindowName(),
                widget.Chord(
                    chords_colors={
                        "launch": ("#ff0000", "#ffffff"),
                    },
                    name_transform=lambda name: name.upper(),
                ),
                widget.TextBox("v0.11.6-alpha", name="default"),
                # NB Systray is incompatible with Wayland, consider using StatusNotifier instead
                widget.StatusNotifier(),
                #widget.Systray(),

                # Keyboard layout
                widget.KeyboardLayout(configured_keyboards=['us','ru']),

                widget.Clock(format="%Y-%m-%d %a %I:%M %p"),
                widget.QuickExit(),
            ],
            24,
            # border_width=[2, 0, 2, 0],  # Draw top and bottom borders
            # border_color=["ff00ff", "000000", "ff00ff", "000000"]  # Borders are magenta
        ),
        )
]


# Set hooks (events)
@hook.subscribe.startup_once # Call one time on startup (not running on config reload)
def autostart_one_time():
    # Core (need to boot early)
    shell('gentoo-pipewire-launcher &')                     # For sound
    
    # Apps (no matter when to boot)
    shell('~/AppImages/gearlever_filen_1865bf.appimage &')  # For cloud sync
    shell('darkman run &')                                  # For dynamic dark theme
    shell('/usr/lib64/libexec/kdeconnectd &')               # For smartphone integration
    shell('syncthing --no-browser &')                       # For smartphone sync
    shell('swww-daemon &')                                  # For wallpapers

    # User scripts
    spr('sh ~/Scripts/CronTriggered &', shell=True)         # For pywall on every chron event
```

### Required

- [X] I have searched past issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by jwijenbergh** [comment]
Is it for every game?

**Comment by SDD701** [comment]
> Is it for every game?

Yes, at least for every game I've tested (around 6). On other WM (like hyprland) the same games are starting fullscreen, so I guess it's a WM bug. Keybinding for toggling fullscreen on a window is working fine for a game window too

**Comment by SDD701** [comment]
Probably the fullscreen request from a window is not handling right or even there is no such request. I have no experience with wayland or wlroots programming at all, but I probably could help trace the error

**Comment by jwijenbergh** [comment]
Fixed in #4902
